extensions/query-history/src/telemetry.ts

```diff
@@ -0,0 +1,87 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import AdsTelemetryReporter, { TelemetryEventMeasures, TelemetryEventProperties } from '@microsoft/ads-extension-telemetry';
+
+export interface PackageInfo {
+	name: string;
+	version: string;
+	aiKey: string;
+}
+
+const packageInfo = require('../package.json') as PackageInfo;
+
+export const TelemetryReporter = new AdsTelemetryReporter(packageInfo.name, packageInfo.version, packageInfo.aiKey);
+
+/**
+ * A helper class to send an Action event with a duration, timer starts on construction and ends when send() is called.
+ */
+export class TimedAction {
```

Comment:
I'll probably add this to the ads-extension-telemetry package after I finish up with this stuff.